media/js/base/stub-attribution.js

```diff
@@ -336,8 +336,9 @@ if (typeof window.Mozilla === 'undefined') {
 
     StubAttribution.hasValidData = function(data) {
         if (typeof data.utm_content === 'string' && typeof data.referrer === 'string') {
-            // If RTAMO data does not originate from AMO, drop attribution (Issue 10337).
-            if ((/^rta:/).test(decodeURIComponent(data.utm_content)) && data.referrer.indexOf('https://addons.mozilla.org') === -1) {
+            // If RTAMO data does not originate from AMO, drop attribution (Issues 10337, 10524).
+            if (((/^rta:/).test(data.utm_content) || (/^rta%/).test(data.utm_content)) &&
```

Comment:
I am sorry to come back to that again but, having talked to @diox about it, we think we must decode the `utm_content` value in a loop (like FF does) and probably limit the length of the value to avoid decoding that value for a very long time (although that is more a perf optimization than an actual issue).

The reason is: `rta:` can also be encoded like this `%72%74%61%3A` and FF will decode it fully and accept it (because the decoded value will start with `rta:`).

Attribution "code" in FF is [limited to 1010](https://searchfox.org/mozilla-central/source/browser/components/attribution/AttributionCode.jsm#143) and AFAIK that includes all the parameters/values. For RTAMO specifically, the length of the `content` value should not be more than `120` characters. It could be more if we don't want to possibly break other stuff (non-AMO stuff). I suppose a limit of 200 chars would already be good because that would already avoid decoding the value for too long but, then again, that is up to you.

Reply:
How about we simply drop the fully encoded rta: value like in your example here?

I’m not sure I agree it’s worth the effort of doing all the decoding logic tbh.

Reply:
Ok, I took a look in GA and the longest `utm_content` value I can see (in the last 30 days) was ~100 chars and was from AMO. With that in mind, I don't think I'm quite as concerned about potentially dropping data.

Let me see if i can rework this again to accomodate decoding.

Reply:
> Fully encoded rta: was just an example, there could be any number of chars encoded, and each could be encoded more than once.

I understand that, my question was more around if the juice is really worth the squeeze here. I did try and assess our current GA data for `utm_content` in the comment above. With that in mind, I think I'm ok with potentially dropping longer values.

Reply:
FWIW the maximum length AMO should send is 111 (Add-on ID is 80 chars max, everything ascii, base64 encoded without the padding that should be 107, add 4 chars for `rta:`). Will suggested 120 above to be safe.

Reply:
I've settled on 150 just to give some extra wiggle room 👍